File: CalculatorAndroid/app/src/main/java/com/ilya/calculatorandroid/MainActivity.java
package com.ilya.calculatorandroid;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    private Button button;
    private EditText aInput;
    private EditText op;
    private EditText bInput;
    private EditText result;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        aInput = (EditText) findViewById(R.id.aInput);
        op = (EditText) findViewById(R.id.op);
        bInput = (EditText) findViewById(R.id.bInput);
        result = (EditText) findViewById(R.id.result);
        button = (Button) findViewById(R.id.button);
    }

    public void onClickCalculate(View v) {
        String oper;
        int a;
        int b;
        String res;

        oper = op.getText().toString();
        a = Integer.parseInt(aInput.getText().toString());
        b = Integer.parseInt(bInput.getText().toString());
        res = Integer.toString(calculate(a, b, oper));
        result.setText(res);
    }

    private int calculate(int a, int b, String op) {
        int res = 0;

        switch (op){
            case "+":
                res = a + b;
                break;
            case "-":
                res = a - b;
                break;
            case "*":
                res = a * b;
                break;
            case "/":
                if ((a == 0) && (b == 0)) throw new ArithmeticException("NaN");
                if (b == 0) throw new ArithmeticException("Divide by zero");
                res = a / b;
                break;
            default:
                res = -1;
                break;
        }

        return res;
    }
}

File: CalculatorXamarin/CalculatorXamarin/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;
using Android.Runtime;
using Android.Views;
using System;

namespace CalculatorXamarin
{
    [Activity(Label = "CalculatorXamarin", MainLauncher = true, Icon = "@drawable/icon")]
    public class MainActivity : Activity
    {
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            // Set our view from the "main" layout resource
            SetContentView (Resource.Layout.Main);

            Button btnCalc = FindViewById<Button>(Resource.Id.calc);
            EditText a = FindViewById<EditText>(Resource.Id.aInput);
            EditText b = FindViewById<EditText>(Resource.Id.bInput);
            EditText res = FindViewById<EditText>(Resource.Id.res);
            EditText op = FindViewById<EditText>(Resource.Id.op);

            btnCalc.Click += delegate
            {
                int n1 = Convert.ToInt32(a.Text);
                int n2 = Convert.ToInt32(b.Text);
                string operation = op.Text;

                Calculation c = new Calculation();
                res.Text = Convert.ToString(c.Calculate(n1, n2, operation));

            };
        }
    }
}

